models/cycle_gan_semantic_model.py

```diff
@@ -158,52 +94,30 @@ def __init__(self, opt):
                 for loss_name in self.loss_names:
                     setattr(self, "loss_" + loss_name, 0)
             
-            self.niter=0
-            
     def set_input(self, input):
-        AtoB = self.opt.direction == 'AtoB'
-        self.real_A = input['A' if AtoB else 'B'].to(self.device)
-        self.real_B = input['B' if AtoB else 'A'].to(self.device)
-        self.image_paths = input['A_paths' if AtoB else 'B_paths']
-        #print(input['B'])
+        super().set_input(input)
         if 'A_label' in input:
-            #self.input_A_label = input['A_label' if AtoB else 'B_label'].to(self.device)
             if not self.opt.regression:
                 self.input_A_label = input['A_label'].to(self.device)
             else:
                 self.input_A_label = input['A_label'].to(torch.float).to(device=self.device)
-            #self.input_B_label = input['B_label' if AtoB else 'A_label'].to(self.device) # beniz: unused
-            #self.image_paths = input['B_paths'] # Hack!! forcing the labels to corresopnd to B domain
         if 'B_label' in input:
             if not self.opt.regression:
                 self.input_B_label = input['B_label'].to(self.device)
             else:
                 self.input_B_label = input['B_label'].to(torch.float).to(device=self.device)
                 
-
-
     def forward(self):
-        self.fake_B = self.netG_A(self.real_A)
-
+        super().forward()
         if self.rec_noise > 0.0:
             self.fake_B_noisy1 = gaussian(self.fake_B, self.rec_noise)
             self.rec_A= self.netG_B(self.fake_B_noisy1)
-        else:
```

Comment:
Where did the `else` go?

Reply:
this behavior is alreay implemented in `CycleGANModel` !